Xiao Improve new-user Team and Project setup efficiency

AuroraLovesWhales opened the pull request:
# Description
1. (PRIORITY MEDIUM) Jae: Improve new-user Team and Project setup efficiency
    1. Admin/Owner Login → Other Links → User Management → Create New User
    2. Right now, when I create a new user I have more clicks than I need for “Assign Project” and “Assign Team”
    3. What I’d like instead of what is shown below is to change the “Project” and “Team” tabs into “Assign Project’ and “Assign Team” buttons with the functions those buttons currently perform… eliminating the need for clicking the tabs and moving the information in those tabs so it is immediately visible with scrolling. 
![unknown](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/6b51e5dc-911a-46a5-9477-3cf7b8520b35)

## Main changes explained:
Removed nav component in UserProfileAdd.jsx and refactored the code to make "Assign Team" appear directly visible under "Assign Project".

## How to test:
    1. check into current branch
    2. do npm run start:local to run this PR locally
    3. log in as Admin/Owner, go to Other Links → User Management → Create New User
    5. verify "Assign Team" is immediately visible with scrolling

## Screenshots or videos of changes:
After:
<img width="695" alt="Pasted Graphic" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/a65357af-3202-4d24-9f9f-321b22ee2c1a">

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/063c341b-8569-4ca7-8df8-31081bf01679

## Note:
Test to make sure the functionalities for "Assign Project" and "Assign Team" remain unchanged.

one-community commented:
Looks good! Please fix all the alignment on the right: 
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/18335122/0058881b-6106-485e-8b8a-03d87a49598c)

Delete buttons should be aligned right and the blue buttons should line up on the right too.

stevenpaalz reviewed (changes requested):
Delete buttons are misaligned as stated in a prior comment. Also a scroll is needed to see these buttons, however I think that is inevitable with the size of the form.

Other than misaligned buttons, this is working as intended. Confirmed that correct functionality of both buttons still works.
<img width="764" alt="Screenshot 2023-05-26 at 4 22 03 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/121642083/6b9dd1f8-0dcf-4b5a-b53e-a848f650f24a">

NidaZaki reviewed (changes requested):
Tested this PR and it works as intended. The functionality for both the buttons are correct. 

![PR#850](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/32549172/8a7d3ce4-0c46-40f0-9373-90b2b4ec6451)

As suggested in previous comments, the delete button for the team can be aligned to the right.

AuroraLovesWhales commented:
I have adjusted the Delete button for Teams. I also added some padding to the "All", "Active", and "Complete" buttons to Tasks. However, I am unable to make the blue buttons ("Assign Project" and "Assign Team") line up on the right. Would appreciate some input on this. Thanks! Now the UI looks like this:
![Pasted Graphic 1](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/dce3013f-4a51-4ed0-89c6-c2047c5cfe67)

stevenpaalz reviewed (dismissed):
Delete button updates looks better. Thanks.

OAkinbode reviewed (changes requested):
Hi, This idea and implementation look really good. I just have one question on this. What is that "Task" bar meant for when adding a new user?

<img width="1440" alt="New user process" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/ea49c5de-9354-46fb-85ab-02ec48a51cc4">

I added a team and project I created that has a task assigned to the current account (sola test - admin). If tasks are attached primarily to individuals, will it be useful to create a task here and attach it to the new user? And if this will clog the page (I think it will) then we should drop the task section from the page altogether.
<img width="730" alt="My team and project" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/21c628dc-c9a0-435f-b5d0-808dc8a2885f">

I may be looking at this all wrong. So please correct me if I am wrong. 

Cheers,

EdwinLauM reviewed (dismissed):
Works as intended! Suggestions were implemented as well! Good job! 

<img width="765" alt="Screen Shot 2023-05-26 at 3 57 13 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/42288987/427e1237-19a0-46d2-892a-355d9e6146f5">

AuroraLovesWhales commented:
> Hi, This idea and implementation look really good. I just have one question on this. What is that "Task" bar meant for when adding a new user?
> 
> <img alt="New user process" width="1440" src="https://user-images.githubusercontent.com/104801652/241336897-ea49c5de-9354-46fb-85ab-02ec48a51cc4.png">
> I added a team and project I created that has a task assigned to the current account (sola test - admin). If tasks are attached primarily to individuals, will it be useful to create a task here and attach it to the new user? And if this will clog the page (I think it will) then we should drop the task section from the page altogether. <img alt="My team and project" width="730" src="https://user-images.githubusercontent.com/104801652/241337232-21c628dc-c9a0-435f-b5d0-808dc8a2885f.png">
> 
> I may be looking at this all wrong. So please correct me if I am wrong.
> 
> Cheers,

Hi @OAkinbode! Thank you for your review. Actually I am not sure about the purpose of the Task section here. It is out of the scope of this PR. Maybe @one-community could confirm this?

yufeiStella reviewed (approved):
This is the safari version. The "Assign Team" appears as intended.
<img width="1296" alt="Screen Shot 2023-05-26 at 23 19 26" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111897624/ae6786d1-b731-45cf-b6df-018a1dbfc66f">

one-community commented:
Good catch, yes, please remove the Tasks section. There is no point in showing that here.

wang9hu commented:
Looking good! The Assign Team button and delete button work as intended! The new user will have the team assigned when created! Good Job!
![Screenshot 2023-05-27 193456](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/77769556/801c1804-49ce-41e5-8cf3-f53e2e14c3c4)
![Screenshot 2023-05-27 193537](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/77769556/4ad94063-28df-4e9e-a304-a2cc3277432b)

20chen-7 reviewed (changes requested):
"style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}" has appeared six times, maybe you should remove duplicated codes. The 'Task' is unnecessary, like previous suggestions, remove this part. Every function is intended.  Great jobs!
<img width="600" alt="Screenshot 2023-05-30 at 12 24 59 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/5ca7cdea-9c66-40ab-93de-44e313f83a24">

papiasharmin reviewed (dismissed):
All functionalities  working as  intended. Great work.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/746a7885-3314-4d2a-b29d-fffedca9366c

NidaZaki commented on src/components/UserProfile/TeamsAndProjects/UserProjectsTable.jsx:
```diff
@@ -294,7 +297,10 @@ const UserProjectsTable = React.memo(props => {
-              <Col md="5" style={{ display: 'flex', alignItems: 'center', justifyContent: 'center'}}>
+              <Col
+                md="5"
+                style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}
```
As pointed in one of the previous comments, maybe create a separate class with common styles so it is reusable.

AnishPandita commented:
Hi @AuroraLovesWhales I tested your PR and the functionalities work as intended. The padding and indentation of the buttons also looks much cleaner and aligned. Great Job! Also i agree with the comments above to remove the duplicate code lines for 'style' tags.
<img width="1439" alt="Screenshot 2023-05-31 at 7 45 40 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/111321945/071bce84-bbe6-41d3-9d32-2832f3114f72">

lchua2314 reviewed (changes requested):
Hi @AuroraLovesWhales I have three changes I would like you to do before I approve this PR:
1. Abstract the repeated code like what the others have requested.
2. Remove the Tasks section like what others have requested.
3. Please attempt to fix the package-lock.json if you're not making any changes with the dependencies. 
 
While I do request changes, the core functionality of the new feature works as intended. Good job!
![Screenshot (2486)](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/46981223/c3ef504a-4102-4c69-a216-4c03ed5491a2)

eduhorta commented on src/components/UserProfile/TeamsAndProjects/UserProjectsTable.jsx:
```diff
@@ -332,7 +338,13 @@ const UserProjectsTable = React.memo(props => {
+                            style={{
+                              display: 'flex',
+                              alignItems: 'center',
+                              justifyContent: 'center',
```
As suggested by other members, how about creating a separate class to facilitate reusability of this style?

eduhorta reviewed (approved):
It works as intended! Nice job! 👍

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/3396f87d-89f2-469c-9419-bf4280f1a7d5

Lucas-E reviewed (dismissed):
I've reviewed your pull request
![Captura de tela de 2023-06-03 16-28-24](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92558269/05a5e2cb-bf19-4f0b-9b05-e45426c5a933)
 and I'm happy to confirm that it works as intended.

AuroraLovesWhales commented:
> Looking good! The Assign Team button and delete button work as intended! The new user will have the team assigned when created! Good Job! Also since the toggle functionality is not used anymore, I think it is better to remove unused state and method like `activeTab`, `toggleTab()`. ![Screenshot 2023-05-27 193456](https://user-images.githubusercontent.com/77769556/241471234-801c1804-49ce-41e5-8cf3-f53e2e14c3c4.png) ![Screenshot 2023-05-27 193537](https://user-images.githubusercontent.com/77769556/241471319-4ad94063-28df-4e9e-a304-a2cc3277432b.png)

Hi @wang9hu! Thank you for your review. I have addressed your suggestions of removing `activeTab`, `toggleTab()`

AuroraLovesWhales commented:
> "style={{ display: 'flex', alignItems: 'center', justifyContent: 'center' }}" has appeared six times, maybe you should remove duplicated codes. The 'Task' is unnecessary, like previous suggestions, remove this part. Every function is intended. Great jobs! <img alt="Screenshot 2023-05-30 at 12 24 59 PM" width="600" src="https://user-images.githubusercontent.com/102696148/242062724-5ca7cdea-9c66-40ab-93de-44e313f83a24.png">

Hi @20chen-7! Thank you for your review. I have addressed your suggestions of removing duplicated codes in styling and the 'Task' section
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/c4fa851e-fc44-43ca-8da3-6df12aa6b5fc)

20chen-7 reviewed (approved):
Now, it 
<img width="518" alt="Screenshot 2023-06-06 at 6 55 46 PM" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/60b331fb-a8bf-4d99-90f5-7464d8cfd42e">
is really cool!

lchua2314 reviewed (approved):
Looks good now. Good job!

NidaZaki reviewed (approved):
Changes look good to me. Nice work!
![image](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/32549172/850b8619-0a15-49c7-80d1-a0f3a1e0174a)

one-community commented:
Merged this because it had 4 approvals and I've got an urgent bug related to this function that I'm hoping this will fix.
